test/langtools/tools/javac/patterns/NewCaseStructureTest.java

```diff
@@ -0,0 +1,457 @@
+                           case E1(): System.err.println(); yield 0;
+                           default: yield 0;
+                       };
+                   }
+               }
+               record E1() {}
+               """);
+        doTest(base,
+               """
+               package test;
+               public class Test {
+                   private int test(Object o) {
+                       return switch (o) {
+                           case String s: System.err.println();
+                           default: System.err.println(); yield 0;
+                       };
+                   }
+               }
+               """);
+        doTest(base, //XXX: run
+               """
+               package test;
+               public class Test {
+                   private void test(Object o) {
+                       switch (o) {
+                           case String s:
+                           case Integer i:
+                           case Object obj:
+                       }
+                   }
+               }
+               """);
+        doTest(base,
+               """
+               package test;
+               public class Test {
+                   private void test(Object o) {
+                       switch (o) {
+                           case Object obj: System.err.println();
+                           case null: System.err.println(obj);
+                       }
+                   }
+               }
+               """,
+               "Test.java:6:43: compiler.err.cant.resolve.location: kindname.variable, obj, , , (compiler.misc.location: kindname.class, test.Test, null)",
+               "1 error");
+
+        doTest(base,
+               """
+               package test;
+               public class Test {
+                   private int test(Integer o) {
+                       return switch (o) {
+                           case default -> 0;
+                           case 0 -> 0;
+                       };
+                   }
+               }
+               """,
+               "Test.java:5:18: compiler.err.default.label.not.allowed",
+               "1 error");
+    }
+
+    @Test
+    public void testDominance(Path base) throws Exception {
+        //A case label with a case pattern p (guarded or unguarded) dominates another case label with a case constant c if p dominates c, which is defined as follows:
+        // A type pattern that declares a pattern variable of type T dominates a constant c of a primitive type P if the wrapper class of P ([5.1.7]) is a subtype of the erasure of T.
+        doTest(base,
+               """
+               package test;
+               public class Test {
+                   private int test(Integer o) {
+                       return switch (o) {
+                           case Integer i when i > 0 -> 0;
+                           case 0 -> 0;
+                           case Integer i -> 0;
```

Comment:
Well, omitting the unconditional case would make the switch non-exhaustive. While that would not be a blocker for a testcase that verifies that error are reported, I think it is cleaner to avoid unrelated errors in the source code.